src/editors/_shared/hooks/DataTypes/AbstractDataType.js

```diff
@@ -1,160 +1,189 @@
+  getEditComponent() {
+    return this.editComponent.bind(this);
   }
 
-  /**
-   * Get data type label
-   * @returns
-   */
-  getLabel() {
-    return this.label;
+  editComponent(props) {
+    if (this._stringOutput)
+      return this.stringEditComponent(props);
+
+    return this.realEditComponent(props);
   }
 
   /**
-   * Get data type edit component
-   * @returns
+   * parsing strings into real objects
    */
-  getEditComponent() {
-    return this.editComponent;
+  parse(value) {
+    if (value === '')
+      return undefined;
+    try {
+      return JSON.parse(value);
+    } catch (e) {
+      return null; // null is invalid
+    }
   }
 
   /**
-   * Abstract validation : Should fail if not implemented in extended class
-   * @returns
+   * unparsing real objects into strings
    */
-  validate() {
-    // To be implemented in extended class
-    console.warn(
-      "debug validation method not implemented for data type",
-      this.key
-    );
-    return Promise.resolve({
-      success: false,
-      error: "ValidationMethodNotImplemented"
-    });
+  unparse(value) {
+    return typeof(value) === "string" ? value : JSON.stringify(value);
   }
 
   /**
-   * Get Default data type value
-   * @param {*} options
-   * @returns {any} Default value
+   * Get Data type key
    */
-  getDefault(options) {
-    return this.default;
+  getKey() {
+    return this.key;
   }
 
-  //========================================================================================
-  /*                                                                                      *
-   *                                    Private Methods                                   *
-   *                                                                                      */
-  //========================================================================================
-
   /**
-   * Expose lodash toString method
-   * @param {*} value
-   * @returns {string} String value
+   * Get data type label
    */
-  toString(value) {
-    return _toString(value);
+  getLabel() {
+    return this.label;
   }
 
   /**
-   * @private Check if value is already parsed, or if it still needs to be parsed to return
-   * @param {*} value : Value to be parsed
-   * @returns parsed value
+   * Abstract validation : validation for simple types
    */
-  getParsedValue(value) {
-    return typeof value === DATA_TYPES.STRING ? JSON.parse(value) : value;
+  _validate(value) {
+    return value === undefined || typeof value === this.key;
   }
 
   /**
-   *
-   * @param {*} props
-   * @param {*} mode
-   * @returns
+   * Abstract validation : validation for simple types
    */
-  defaultStringEditor(props, mode = "row") {
-    const editor = {
-      row: _props => this.stringEditComponent(_props, ""),
-      dialog: this.codeEditComponent
-    };
-    return editor[mode](props);
+  async validate(value) {
+    // "None" indicates a disabled value
+    if (value === "None")
```

Comment:
```suggestion
    if (value === DISABLED_VALUE)
```
This way you can also remove the comment above.